Runtime/Scripts/MediaStreamTrack.cs

```diff
@@ -25,31 +26,159 @@ public bool Enabled
                 NativeMethods.MediaStreamTrackSetEnabled(self, value);
             }
         }
+
+        /// <summary>
+        /// 
+        /// </summary>
         public TrackState ReadyState
         {
             get
             {
                 return NativeMethods.MediaStreamTrackGetReadyState(self);
             }
-            private set { }
         }
 
-        public TrackKind Kind { get => kind; private set { } }
-        public string Id { get => id; private set { } }
+        /// <summary>
+        /// 
+        /// </summary>
+        public TrackKind Kind { get; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public string Id { get; }
 
         internal MediaStreamTrack(IntPtr ptr)
         {
             self = ptr;
-            kind = NativeMethods.MediaStreamTrackGetKind(self);
-            id = Marshal.PtrToStringAnsi(NativeMethods.MediaStreamTrackGetID(self));
+            WebRTC.Table.Add(self, this);
+            Kind = NativeMethods.MediaStreamTrackGetKind(self);
+            Id = Marshal.PtrToStringAnsi(NativeMethods.MediaStreamTrackGetID(self));
+        }
+
+        ~MediaStreamTrack()
+        {
+            this.Dispose();
+            WebRTC.Table.Remove(self);
+        }
+
+        public virtual void Dispose()
+        {
+            if (this.disposed)
+            {
+                return;
+            }
+            if (self != IntPtr.Zero && !WebRTC.Context.IsNull)
+            {
+                WebRTC.Context.DeleteMediaStreamTrack(self);
+                self = IntPtr.Zero;
+            }
+            this.disposed = true;
+            GC.SuppressFinalize(this);
         }
+
         //Disassociate track from its source(video or audio), not for destroying the track
         public void Stop()
         {
             stopTrack(this);
         }
     }
 
+    public class VideoStreamTrack : MediaStreamTrack
+    {
+        internal static List<VideoStreamTrack> tracks = new List<VideoStreamTrack>();
+
+        bool needFlip = false;
+        UnityEngine.Texture source;
+        UnityEngine.RenderTexture dest;
+
+        private static UnityEngine.RenderTexture CopyRenderTexture(UnityEngine.RenderTexture source)
```

Comment:
`CreateRenderTexture` would be easier to understand if do not perform the RenderTexture copy process here.

Reply:
I fixed it